lib/metadataTypes/MetadataType.js

```diff
@@ -354,18 +352,24 @@ class MetadataType {
                 Util.metadataLogger('error', this.definition.type, 'upsert', ex, metadataKey);
             }
         }
+        const createLimit = pLimit(10);
```

Comment:
what is this doing?

Reply:
This is essentially the rate limiter object, which is applied to each of the iterables. In this case it is limiting it to 10 promises at any given time